app/scripts/directives/annotationReport.js

```diff
@@ -0,0 +1,108 @@
+                            acc.THERAPEUTIC_IMPLICATIONS.push(evidence);
+                        } else {
+                            if (_.endsWith(evidenceType, 'IMPLICATION')) {
+                                evidenceType = evidenceType + 'S';
+                            }
+                            if (!acc[evidenceType]) {
+                                acc[evidenceType] = [];
+                            }
+                            acc[evidenceType].push(evidence);
+                        }
+                        return acc;
+                    }, {}), scope.shownSections);
+                }
+
+                function initializeReportDat() {
+                    scope.reportData = {
+                        tumorTypes: [],
+                        allEvidences: []
+                    };
+
+                }
+
+                initializeReportDat();
+                scope.summaryInfo = [];
+                scope.shownSections = ['THERAPEUTIC_IMPLICATIONS', 'DIAGNOSTIC_IMPLICATIONS', 'PROGNOSTIC_IMPLICATIONS'];
+
+                scope.$watch('data', function() {
+                    initializeReportDat();
+                    scope.reportData.allEvidences = convertEvidence(_.reduce(scope.data.tumorTypes, function(acc, next) {
+                        acc = acc.concat(next.evidences);
+                        return acc;
+                    }, []));
+
+                    _.each(scope.data.tumorTypes, function(tumorType) {
+                        var copyTT = _.cloneDeep(tumorType);
+                        copyTT.evidences = convertEvidence(copyTT.evidences);
+                        scope.reportData.tumorTypes.push(copyTT);
+                    });
+
+                    scope.summaryInfo = [
+                        scope.data.oncogenic,
+                        scope.data.mutationEffect ? scope.data.mutationEffect.knownEffect : '',
+                        scope.data.highestSensitiveLevel ? processLevel(scope.data.highestSensitiveLevel) : '',
+                        scope.data.highestResistanceLevel ? processLevel(scope.data.highestResistanceLevel) : '',
+                        scope.data.highestDiagnosticImplicationLevel ? processLevel(scope.data.highestDiagnosticImplicationLevel) : '',
+                        scope.data.highestPrognosticImplicationLevel ? processLevel(scope.data.highestPrognosticImplicationLevel) : ''].filter(function(item) {
+                        return !!item;
+                    });
+                    scope.summaries = [
+                        scope.data.geneSummary,
+                        scope.data.variantSummary,
+                        scope.data.tumorTypeSummary,
+                        scope.data.diagnosticSummary,
+                        scope.data.prognosticSummary].filter(function(item) {
+                        return !!item;
+                    });
+                });
+                scope.getEvidenceTypeName = function(evidenceType) {
```

Comment:
This function appear twice in this file and evidencesReport.js. Can you move it in mainUtils.js